sway-core/src/decl_engine/mapping.rs

```diff
@@ -99,4 +102,25 @@ impl DeclMapping {
         }
         None
     }
+
+    pub(crate) fn filter(&self, self_type: TypeId, engines: &Engines) -> DeclMapping {
```

Comment:
Just naming it `filter` seems too vague. Could be appropriate if it's not ambiguous the sorts of filtering you can do on a `DeclMapping` but it's it's not readily apparent to me.

At the very least this needs a doc comment explaining what it's doing.